onnxruntime/core/framework/error_code.cc

```diff
@@ -7,39 +7,37 @@
 #include <cassert>
 using onnxruntime::common::Status;
 
+struct OrtStatus {
+  OrtErrorCode code;
+  char msg[1];  // a null-terminated string
+};
+
 ORT_API(OrtStatus*, OrtCreateStatus, OrtErrorCode code, const char* msg) {
   assert(!(code == 0 && msg != nullptr));
   size_t clen = strlen(msg);
-  size_t len = clen + 1 + sizeof(int);
-  char* p = new char[len];
-  char* ret = p;
-  *reinterpret_cast<int*>(p) = static_cast<int>(code);
-  p += sizeof(int);
-  memcpy(p, msg, clen);
-  p += clen;
-  *p = '\0';
-  return ret;
+  OrtStatus* p = reinterpret_cast<OrtStatus*>(new char[sizeof(OrtStatus) + clen]);
```

Comment:
What if sizeof(OrtStatus) is larger than you expected?
The new code looks beautiful, but it's not portable.

Reply:
If sizeof(OrtStatus) is larger due to alignment everything is still fine. How would this break?

Reply:
Either we hide the memory layout, don't tell people what's inside an OrtStatus. Or we describe it in the header file, and implement it exactly as we described. Previously, I chose the second. Because OrtStatus would be used extensively, performance matters.  We may put OrtGetErrorCode and OrtGetErrorMessage into a header file and inline these functions, but a non-inlined version is still needed for C# users.